### PR TITLE
update: changes of consultation 1

### PR DESCRIPTION
**Pull request by dev-laky**
This pull request introduces several significant updates to the `peakpoker` project, including improvements to type safety, code refactoring, and enhancements to the test suite. Key changes include replacing string-based card suits and ranks with enums, renaming enums and files for consistency, refactoring the `Deck` class for better initialization and functionality, and updating tests to align with the new implementations.

### Type Safety Improvements:
* Updated `Card` class to use `Suit` and `Rank` enums instead of strings for `suit` and `rank` properties, improving type safety (`src/main/kotlin/hwr/oop/projects/peakpoker/core/card/Card.kt`).
* Replaced string-based suits and ranks in `Deck` initialization with `Suit.values()` and `Rank.values()` enums, ensuring consistency and reducing potential errors (`src/main/kotlin/hwr/oop/projects/peakpoker/core/deck/Deck.kt`).

### Enum and File Renaming:
* Renamed `GameActions` to `GameAction`, `GameStates` to `GameState`, and `PlayerRoles` to `PlayerRole` for better naming consistency. Updated all references across the codebase, including database entities and tables.

### `Deck` Class Refactoring:
* Refactored `Deck` class to initialize cards using enums and removed redundant shuffle logic. Added a `draw` method for drawing cards and tracking dealt cards, and renamed `peak` to `show` for better clarity.

### Test Suite Enhancements:
* Replaced `assertThat` with `assertEquals` and `assertThrows` in tests for better alignment with the JUnit 5 framework. Added new tests for `Deck` to verify initialization, card drawing, and exception handling when the deck is empty (`src/test/kotlin/hwr/oop/projects/peakpoker/DeckTest.kt`).
* Updated `PlayerTest` to use the renamed `getBet` property (`src/test/kotlin/hwr/oop/projects/peakpoker/PlayerTest.kt`).
This pull request introduces several updates to the `peakpoker` project, focusing on improving type safety, renaming for consistency, refactoring the `Deck` class, and enhancing test coverage. The most significant changes include replacing string-based enums with strongly-typed enums, refactoring the `Deck` class for better initialization and functionality, and renaming files and entities for clarity.

### Type Safety Improvements:
* Replaced `String` values for `Card`'s `suit` and `rank` with strongly-typed `Suit` and `Rank` enums in `Card` (`src/main/kotlin/hwr/oop/projects/peakpoker/core/card/Card.kt`).

### Class and File Refactoring:
* Refactored the `Deck` class to use `Suit` and `Rank` enums for initialization, removed redundant shuffle logic, and added a `draw` method to handle card drawing (`src/main/kotlin/hwr/oop/projects/peakpoker/core/deck/Deck.kt`).
* Renamed files and enums for consistency, e.g., `GameActions` to `GameAction`, `GameStates` to `GameState`, and `PlayerRoles` to `PlayerRole` across multiple files.

### Database Schema Updates:
* Updated database-related files to use the renamed enums (`GameAction`, `GameState`, `PlayerRole`) in place of the old ones, ensuring consistency across the codebase (`src/main/kotlin/hwr/oop/projects/peakpoker/db/entities`, `src/main/kotlin/hwr/oop/projects/peakpoker/db/tables`).

### Test Enhancements:
* Added new test cases for the `Deck` class to validate its initialization, `show`, and `draw` methods, including edge cases like drawing from an empty deck (`src/test/kotlin/hwr/oop/projects/peakpoker/DeckTest.kt`).
* Updated existing tests to reflect changes in method names, e.g., `currentBet` to `getBet` in `PlayerTest` (`src/test/kotlin/hwr/oop/projects/peakpoker/PlayerTest.kt`).

### License Update:
* Updated the `LICENSE` file to include the names of contributors for 2024 (`LICENSE`).
### Miscellaneous:
* Updated `LICENSE` file with proper copyright information for 2024 (`LICENSE`).
* Removed outdated TODO comments and improved code readability in `Player` and other classes (`src/main/kotlin/hwr/oop/projects/peakpoker/core/player/Player.kt`).
This pull request introduces significant updates to the `peakpoker` project, including improvements to the card and deck system, renaming and standardization of enums, updates to database entities and tables, and enhancements to tests. These changes aim to improve code clarity, maintainability, and functionality.

### Core Changes to Card and Deck System:
* Updated `Card` class to use `Suit` and `Rank` enums instead of strings for `suit` and `rank`, improving type safety (`src/main/kotlin/hwr/oop/projects/peakpoker/core/card/Card.kt`).
* Refactored the `Deck` class to initialize cards using `Suit` and `Rank` enums, removed redundant shuffle logic, and added a `draw` method to handle card drawing with error handling (`src/main/kotlin/hwr/oop/projects/peakpoker/core/deck/Deck.kt`).

### Enum Renaming and Standardization:
* Renamed `GameActions` to `GameAction`, `GameStates` to `GameState`, and `PlayerRoles` to `PlayerRole` for consistency. Updated all references across the codebase, including database entities and tables

### Database Updates:
* Updated database entities (`BetEntity`, `GameEntity`, `GamePlayerEntity`) and tables (`Bets`, `Games`, `GamePlayers`) to reflect the renamed enums and improve consistency in naming.

### Test Enhancements:
* Added new tests for the `Deck` class to verify deck initialization, card drawing, and exception handling when the deck is empty. Updated `PlayerTest` to reflect changes in getter naming.

### Licensing:
* Updated the `LICENSE` file to include the names of contributors for 2024 (`LICENSE`).

### CODE REVIEW

File: src/test/kotlin/hwr/oop/projects/peakpoker/DeckTest.kt [hunk 1]
```diff
@@ -2,22 +2,60 @@ package hwr.oop.projects.peakpoker
...
+        val cards = deck.show()
 
         // then
-        val differentPositions = deckBefore.zip(deckAfter).count { (before, after) -> before != after }
-        assertThat(differentPositions).isGreaterThan(0)
+        assertEquals(52, cards.size)
+    }
+
+    @Test
+    fun `peak returns a copy of the cards`() {
```
Comment (juliankuecken) [inline src/test/kotlin/hwr/oop/projects/peakpoker/DeckTest.kt:24]:
```suggestion
    fun `show returns a copy of the cards`() {
```

File: src/test/kotlin/hwr/oop/projects/peakpoker/DeckTest.kt [hunk 2]
```diff
@@ -2,22 +2,60 @@ package hwr.oop.projects.peakpoker
...
+        assertFalse(deck.show().contains(drawnCard))
+    }
+
+    @Test
+    fun `draw throws exception when no cards left`() {
+        // given
+        val deck = Deck()
+        repeat(52) { deck.draw() } // draw all cards
+
+        // when & then
```
Comment (juliankuecken) [inline src/test/kotlin/hwr/oop/projects/peakpoker/DeckTest.kt:55]:
```suggestion
        // when & then
        assertEquals(0, deck.show().size) // Check if the list is actually empty (the size is zero)
```

**Review by juliankuecken — CHANGES_REQUESTED** [review]
Good code, thanks Louis!

**Review by timwiddo — APPROVED** [review]
LGTM!

**Review by juliankuecken — APPROVED** [review]
